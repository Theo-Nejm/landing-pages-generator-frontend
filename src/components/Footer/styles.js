import styled, { css } from 'styled-components';

export const Container = styled.footer`
  ${({ theme }) => css`
    p,
    a,
    i,
    b {
      font-size: ${theme.fonts.sizes.xxsmall};
      text-align: center;
    }
  `}
`;
