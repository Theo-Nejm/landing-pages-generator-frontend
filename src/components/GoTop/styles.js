import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    color: #fff;
    background: ${theme.colors.primaryColor};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    border-radius: 50%;
    position: fixed;
    bottom: 2rem;
    right: 2rem;
    z-index: 4;
    opacity: 0.85;
  `}
`;
