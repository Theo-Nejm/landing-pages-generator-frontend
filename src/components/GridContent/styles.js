import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    text-align: center;
    max-width: 58rem;
  `}
`;

export const HTMLContainer = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xlarge} 0;
  `}
`;
