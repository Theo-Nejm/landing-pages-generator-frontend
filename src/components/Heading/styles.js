import styled, { css } from 'styled-components';

const titleSize = {
  xsmall: (theme) =>
    css`
      font-size: ${theme.fonts.sizes.xsmall};
    `,
  small: (theme) =>
    css`
      font-size: ${theme.fonts.sizes.small};
    `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,
  large: (theme) =>
    css`
      font-size: ${theme.fonts.sizes.large};
    `,
  xlarge: (theme) =>
    css`
      font-size: ${theme.fonts.sizes.xlarge};
    `,
  xxlarge: (theme) =>
    css`
      font-size: ${theme.fonts.sizes.xxlarge};
    `,
  huge: (theme) =>
    css`
      font-size: ${theme.fonts.sizes.huge};
      ${mediaFont(theme)};
    `,
  xhuge: (theme) =>
    css`
      font-size: ${theme.fonts.sizes.xhuge};
      ${mediaFont(theme)};
    `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.medium} {
    font-size: ${theme.fonts.sizes.large};
  }
`;

const titleCase = (upperCase) => css`
  text-transform: ${upperCase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, upperCase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : '#FFF'};
    ${titleSize[size](theme)};
    ${titleCase(upperCase)};
  `}
`;
