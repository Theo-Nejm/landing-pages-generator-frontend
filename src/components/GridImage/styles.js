import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    h2 {
      font-size: 4.2rem;
    }

    ${TextComponent} {
      margin-bottom: ${theme.spacings.huge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};
  `}
`;

export const Element = styled.div`
  ${() => css`
    overflow: hidden;

    > img {
      width: 100%;
      transition: all 0.3s ease-in-out;

      &:hover {
        transform: scale(1.2) rotate(10deg);
      }
    }
  `}
`;
