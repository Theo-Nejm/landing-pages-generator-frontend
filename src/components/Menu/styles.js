import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';

const menuVisible = () => css`
  visibility: visible;
  opacity: 1;
`;

export const Container = styled.div`
  ${({ theme, isOpened }) => css`
    position: fixed;
    z-index: 5;
    top: 0;
    left: 0;
    right: 0;
    width: 100vw;
    transition: all .2s ease-in-out;
    height: 7rem;
    border-bottom: 1px solid ${theme.colors.mediumGray};

    > ${SectionContainer} {
      padding-top: 0;
      padding-botom: 0;
    }

    @media ${theme.media.medium} {
      height: 100vh;
      visibility: hidden;
      opacity: 0;

      ${isOpened && menuVisible}
    }
  }
  `}
`;

export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: space-between;
    align-items: center;
    height: 7rem;

    @media ${theme.media.medium} {
      display: flex;
      flex-flow: column nowrap;
      text-align: center;
      align-items: center;
      justify-content: center;
      height: 100vh;

      img,
      h1 {
        margin-bottom: 2rem;
      }
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    z-index: 6;
    position: fixed;
    top: 2rem;
    right: 2rem;
    width: 3.5rem;
    height: 3.5rem;
    cursor: pointer;
    background: none;
    border: none;
    color: ${theme.colors.primaryColor};
    display: none;
    @media ${theme.media.medium} {
      display: flex;
    }
  `}
`;
