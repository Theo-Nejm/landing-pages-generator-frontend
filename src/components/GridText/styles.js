import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-flow: column nowrap;
    margin-top: 6rem;

    h2 {
      font-size: 4.2rem;
    }

    ${TextComponent} {
      margin-bottom: ${theme.spacings.huge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.medium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const Element = styled.div`
  ${({ theme }) => css`
    ${Heading} {
      position: relative;
      left: 5rem;
    }

    ${Heading}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: ${theme.fonts.sizes.xhuge};
      left: -5rem;
      top: -2rem;
      transform: rotate(7.5deg);
    }
  `}
`;
