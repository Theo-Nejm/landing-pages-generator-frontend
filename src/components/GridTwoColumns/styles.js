import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 2fr;
    align-items: center;
    gap: ${theme.spacings.large};

    @media ${theme.media.medium} {
      grid-template-columns: 1fr;
      text-align: center;
    }

    ${Title} {
      margin-bottom: ${theme.spacings.medium};
    }
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
  background: none;
  @media ${theme.media.medium} {
    margin-bottom: ${theme.spacings.medium};
  }
    }
  `}
`;

export const ImageContainer = styled.div`
  > img {
    width: 100%;
  }
`;
