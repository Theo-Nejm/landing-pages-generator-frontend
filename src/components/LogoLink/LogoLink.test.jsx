import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { LogoLink } from '.';

describe('<LogoLink />', () => {
  it('should render text logo', () => {
    renderTheme(<LogoLink link="#" text="MyLogo" />);
    const heading = screen.getByRole('heading', { name: 'MyLogo' });
    expect(heading.firstChild).toHaveAttribute('href', '#');
  });

  it('should render image logo', () => {
    renderTheme(<LogoLink link="#" text="MyLogo" srcImg="http://image" />);
    const logo = screen.getByRole('heading', { name: 'MyLogo' });
    expect(logo.firstChild).toHaveAttribute('href', '#');
    expect(logo.firstChild.firstChild).toHaveAttribute('src', 'http://image');
    expect(logo.firstChild.firstChild).toHaveAttribute('alt', 'MyLogo');
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(
      <LogoLink link="#" text="MyLogo" srcImg="http://image" />,
    );
    screen.getByRole('heading', { name: 'MyLogo' });
    expect(container).toMatchSnapshot();
  });
});
