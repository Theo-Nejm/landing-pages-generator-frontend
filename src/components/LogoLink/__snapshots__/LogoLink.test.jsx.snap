// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink /> should match snapshot 1`] = `
.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 > img {
  max-height: 2.4rem;
}

.c0 {
  color: #0a1128;
  font-size: 2.4rem;
  text-transform: uppercase;
}

<div>
  <h1
    class="c0"
  >
    <a
      class="c1"
      href="#"
    >
      <img
        alt="MyLogo"
        src="http://image"
      />
    </a>
  </h1>
</div>
`;
